ckanext/security/auth.py

```diff
@@ -0,0 +1,37 @@
+import ckan.logic as logic
+import ckan.authz as authz
+import ckan.plugins.toolkit as toolkit
+from ckan.lib.base import _
+from ckan.logic.auth import (get_package_object, get_group_object,
+                            get_resource_object)
+
+try:
+    from ckan.lib.plugins import get_permission_labels
+except ImportError:
+    pass
+
+
+def user_list(context, data_dict):
+    # Get the user name of the logged-in user.
+    user = context['user']
+
+    # check if user doesn't exist (e.g. they're not logged-in)
+    convert_user_name_or_id_to_id = toolkit.get_converter('convert_user_name_or_id_to_id')
+    try:
+        user_id = convert_user_name_or_id_to_id(user, context)
+    except toolkit.Invalid:
+        return {'success': False}
+    return {'success': True}
+
```

Comment:
How does the `user_list` function and `user_show` function differ from each other?

Reply:
The user_list auth function is for determining who can view the pages that list all users. The user_show auth function determines who can view individual user profiles.

Reply:
Their definition looks identical. if so, we can dry the code.